python/cudf/cudf/_libxx/strings/translate.pyx

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2018-2020, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+from cudf._libxx.move cimport move
+
+from cudf._libxx.cpp.column.column cimport column
+from cudf._libxx.cpp.column.column_view cimport column_view
+from cudf._libxx.cpp.strings.translate cimport (
+    translate as cpp_translate
+)
+from cudf._libxx.column cimport Column
+from collections import OrderedDict
+from libcpp.vector cimport vector
+from libcpp.pair cimport pair
+from cudf._libxx.cpp.types cimport char_utf8
+
+
+def translate(Column source_strings,
+              object mapping_table):
+    """
+    Translates individual characters within each string
+    mapping present in the mapping_table.
+    """
+    cdef unique_ptr[column] c_result
+    cdef column_view source_view = source_strings.view()
+
+    mapping_table_dict = OrderedDict()
```

Comment:
Why use an `OrderedDict` here if you don't need to do lookups? Can just use a list. Or better yet, create the `c_mapping_table` vector before this loop, reserve the size needed, and `push_back` into the vector from this loop.